ledger/src/shred.rs

```diff
@@ -1,4 +1,54 @@
-//! The `shred` module defines data structures and methods to pull MTU sized data frames from the network.
+//! The `shred` module defines data structures and methods to pull MTU sized data frames from the
+//! network. There are two types of shreds: data and coding. Data shreds contain entry information
+//! while coding shreds provide redundancy to protect against dropped network packets (erasures).
+//!
+//! +---------------------------------------------------------------------------------------------+
+//! | Data Shred                                                                                  |
+//! +---------------------------------------------------------------------------------------------+
+//! | common       | data       | payload                                                         |
+//! | header       | header     |                                                                 |
+//! |+---+---+---  |+---+---+---|+----------------------------------------------------------+----+|
+//! || s | s | .   || p | f | s || data (ie ledger entries)                                 | r  ||
+//! || i | h | .   || a | l | i ||                                                          | e  ||
+//! || g | r | .   || r | a | z || See notes below for explanation of restricted            | s  ||
```

Comment:
Maybe wrap "restricted' in quotes to signal that it's a field in the struct. Also, maybe reference the specific function where this explanation is available?

Reply:
Done - put in quotes and mentioned that the "notes" are immediately after the shred diagrams